keps/sig-api-machinery/3352-aggregated-discovery/README.md

```diff
@@ -641,6 +645,31 @@ feature and upgrade/downgrade is not a problem.
 
 ### Version Skew Strategy
 
+When moving from beta to GA, we will introduce a new API group version `apidiscovery.k8s.io/v2`.
+
+All clients v1.26 to v1.29 will only request for the beta API group version `apidiscovery.k8s.io/v1beta2`.
+
+To accommodate skew between the client and server (older client and newer server), the server will serve both v2 and v2beta1 versions based on the client request headers. The API server will continue to support v2beta1 until its removal in Kubernetes v1.33.
+
+To accommodate skew between an older server and newer client, starting in v1.30,
+client-go will request for both v2 and v2beta1 by sending a list of group versions
+requested (in order from v2, v2beta1, unaggregated) and the server will return the
+first group version that matches. Concretely, this is done using `Accept` headers with a single request.
+
+```
+Accept: application/json;as=APIGroupDiscoveryList;v=v2;g=apidiscovery.k8s.io,application/json;as=APIGroupDiscoveryList;v=v2beta1;g=apidiscovery.k8s.io,application/json
+```
+
+In the case of older servers, the server will only
+be able to match v2beta1. The client will support both v2 and v2beta1. This allows a
+newer client to communicate with an older server that supports only the beta version.
+Other clients should follow the same convention to support version skew, though a client
+that is only capable of processing v2 is sufficient if it only communicates with v1.30+ servers.
+Otherwise, the client will need to be ready to tolerate a 406 Not Acceptable response and handle
+the error appropriately.
+
+If there is no skew and both server and client are v1.30+, clients will still request for v2 and v2beta1, and the server will match the first group version and return v2.
+
 ## Production Readiness Review Questionnaire
```

Comment:
For PRR:

We should include an answer to "What are the reasonable SLOs (Service Level Objectives) for the enhancement?"  In particular, what's the SLO for client requests to aggregated discovery?

For "What steps should be taken if SLOs are not being met to determine the problem?" The feature gate will go away as part of promotion to GA. Once this happens, what steps should be taken?  E.g. if there is a very large number of CRDs, or a single CRD with a very large OpenAPI schema, or similar problems with aggregated API servers, how would someone figure out the the SLO is being violated because of one of these things?

Reply:
Updated this section. I think it's fine to stick with the existing Kubernetes SLO for a non-streaming read-only API call which has p99 < 1s.